recipe-server/normandy/studies/api/v2/fields.py

```diff
@@ -0,0 +1,54 @@
+import json
+import zipfile
+
+from rest_framework.fields import FileField
+
+
+class ExtensionFileField(FileField):
+    default_error_messages = {
+        'not_a_zip': 'Extension file must be zip-formatted.',
+        'not_signed': 'Extension file must be signed.',
+        'no_id': 'Web extensions must have a manifest key "applications.gecko.id".',
+        'bad_manifest': 'Web extension manifest is corrupt.',
+    }
+
+    def to_internal_value(self, data):
+        try:
+            xpi_zip = zipfile.ZipFile(data)
+        except zipfile.BadZipFile:
+            self.fail('not_a_zip')
+
+        if self._is_webext(xpi_zip):
+            self._has_webext_id(xpi_zip)
+        self._is_signed(xpi_zip)
+
+        return super().to_internal_value(data)
+
+    def _is_webext(self, xpi_zip):
+        files = xpi_zip.namelist()
+        if 'manifest.json' not in files or 'bootstrap.json' in files:
```

Comment:
Did you mean `bootstrap.js`?

Reply:
Oops, `bootstrap.js` would be more correct. However, I think this whole check is misguided. I don't have any real reason why a file would have both `manifest.json` and `bootstrap.js` so I'm just going to take this out.